### PR TITLE
preserve data

### PR DESCRIPTION
**Pull request by jaamanu**
in this PR we
Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
On exit from your app save all the data (books, people, and rentals) into JSON files. 
One per type of data  storing and named accordingly
On startup of app loads all data from files.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
-    app.run
+class Main
+  def initialize
+    @app = App.new
+  end
+
+  def actions
     puts
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
```
Comment (julie-ify) [inline main.rb:12]: - [ ] Kindly ensure the book details are displayed. This is a refactor for data storage so the previous functionality is still expected to be there.

  djdye

File: main.rb [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
```
Comment (julie-ify) [inline main.rb:17]: - [ ] Kindly ensure the rental details are listed as well.

  duee

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @jaamanu and @KareemWilson,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -1,197 +1,116 @@
...
-    if @books.empty?
-      puts 'Please insert books first!'
-    else
-      @books.each_with_index do |book, index|
-        puts "#{index}) Title: #{book.title}, Author: #{book.author}}"
-      end
-    end
+  def list_books
+    @books.each { |book| puts book }
   end
```
Comment (DammyShittu) [inline app.rb:39]: - [ ] The issue highlighted by the previous reviewer is still there. When I select option 1 to list the books it just displays the same message in the image from the previous reviewer. You can fix that by change this code to this.

  ```
  def list_books
    @books.each { |book| puts "#{book.title} #{book.author}" }
  end
  ```

File: app.rb [hunk 4]
```diff
@@ -1,197 +1,116 @@
...
 
-  # create a rental
-  def create_rental()
-    # select book
-    puts 'Select a book from the following list by number: '
-    list_all_books
-    book_option = gets.chomp.to_i
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, i| puts "#{i}) #{book}" }
```
Comment (DammyShittu) [inline app.rb]: - [ ] When I select option 5 to create a rental and it lists the books the same issue is present here as well. You can change to this:

  ```
  @books.each_with_index { |book, i| puts "#{i}) #{book.title} #{book.author}" }
  ```

File: app.rb [hunk 5]
```diff
@@ -1,197 +1,116 @@
...
-  # run the program
-  def run
-    puts 'Please choose an option by entering a number: '
-    display_cmd
-    user_option_executer
+  def list_rentals
+    print 'ID of person: '
+    id = gets.chomp
+    puts 'Rentals:'
+    @rentals.each { |rental| puts rental if rental.person.id == id.to_i }
```
Comment (DammyShittu) [inline app.rb]: - [ ] When I want to list the rental for a person the same issue is here also. You can change to this:

  ```
  @rentals.each do |rental|
    puts "#{rental.book.title} was rented by #{rental.person.name} on #{rental.date}" if rental.person.id == id.to_i
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @jaamanu and @KareemWilson,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @jaamanu and @KareemWilson,

✔️ UI is well implemented

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**